R/plot_recruitment.R

```diff
@@ -28,7 +35,36 @@ plot_recruitment <- function(dat,
                              scale_amount = NULL,
                              show_warnings = FALSE,
                              end_year = NULL,
-                             return = "recruitment"){
+                             return = "recruitment",
+                             make_rda = FALSE,
+                             rda_dir = getwd()
+                             ){
+
+  # create plot-specific variables to use throughout fxn for naming and IDing
+  topic_label <- "recruitment"
+
+  # identify whether function generates a figure or table
+  # extract name of function housing id_fxn_output
+  fxn_name <- as.character(match.call()[[1]])
+
+  # if housing fxn's name starts with "plot", return "figure"; else, return "table"
+  fig_or_table <- ifelse(startsWith(fxn_name, "plot"),
+                         "figure",
+                         "table")
+
+  # run write_captions.R if its output doesn't exist
+  if (!file.exists(
+    fs::path(getwd(), "captions_alt_text.csv"))
+    ) {
+    satf::write_captions(dat = dat,
+                         dir = getwd(),
+                         year = end_year)
```

Comment:
This should probably happen after the end_year check in case the user doesn't input an end_year